eos/utils/kmatrix.hh

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2019 Danny van Dyk, Stephan Kürten
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_UTILS_KMATRIX_HH
+#define EOS_GUARD_EOS_UTILS_KMATRIX_HH 1
+
+
+#include <eos/utils/complex.hh>
+#include <eos/utils/exception.hh>
+#include <eos/utils/parameters.hh>
+
+#include <gsl/gsl_blas.h>
+#include <gsl/gsl_linalg.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_complex.h>
+#include <gsl/gsl_complex_math.h>
+#include <array>
+#include <vector>
+#include <memory>
+
+namespace eos
+
+{
+    template <unsigned nchannels_, unsigned nresonances_>
+    class KMatrix
+    {
+
+    public:
```

Comment:
```public``` and ```private``` at one level of indentation; everything else at two levels of indentations in a ```class``` scope.